script/draft-release/run.ts

```diff
@@ -0,0 +1,105 @@
+import { spawn } from '../changelog/spawn'
+import { getLogLines } from '../changelog/git'
+import {
+  convertToChangelogFormat,
+  getChangelogEntriesSince,
+} from '../changelog/parser'
+import { sort as semverSort, SemVer } from 'semver'
+import { getNextVersionNumber } from './version'
+
+const jsonStringify: (obj: any) => string = require('json-pretty')
+
+async function getLatestRelease(excludeBetaReleases: boolean): Promise<string> {
+  const allTags = await spawn('git', ['tag'])
+  let releaseTags = allTags
+    .split('\n')
+    .filter(tag => tag.startsWith('release-'))
+    .filter(tag => tag.indexOf('-linux') === -1)
+    .filter(tag => tag.indexOf('-test') === -1)
+
+  if (excludeBetaReleases) {
+    releaseTags = releaseTags.filter(tag => tag.indexOf('-beta') === -1)
+  }
+
+  const releaseVersions = releaseTags.map(tag => tag.substr(8))
+
+  const sortedTags = semverSort(releaseVersions)
+  const latestTag = sortedTags[sortedTags.length - 1]
+
+  if (latestTag instanceof SemVer) {
+    return latestTag.raw
+  }
+
+  return latestTag
+}
+
+function parseChannel(arg: string): 'production' | 'beta' {
+  if (arg === 'production' || arg === 'beta') {
+    return arg
+  }
+
+  throw new Error(`An invalid channel ${arg} has been provided`)
```

Comment:
You might be able to `assertNever` here.

Reply:
`assertNever` is an application API. I'd rather just have this fail fast than couple this to app code at this point.